assets/src/editor/events/ticket-assignments-manager/ticket-assignments-manager-modal.js

```diff
@@ -637,6 +769,66 @@ export default compose( [
 		};
 		return { addTicketEntities, removeTicketEntities };
 	} ),
+	( WrappedComponent ) => ( {
+		dateEntities,
+		ticketEntities,
+		...otherProps
+	} ) => {
+		const unfilteredDates = dateEntities;
+		const unfilteredTickets = ticketEntities;
+		const [ showArchivedDates, setShowArchivedDates ] = useState( false );
+		const [ showExpiredDates, setShowExpiredDates ] = useState( false );
+		const [ showArchivedTickets, setShowArchivedTickets ] = useState( false );
+		const [ showExpiredTickets, setShowExpiredTickets ] = useState( false );
```

Comment:
When we get this amount of state setters/values, I think it might work a bit better to implement [`useReducer`](https://reactjs.org/docs/hooks-reference.html#usereducer) especially for actions that belong in the same "type" of ui/ux behaviour (which in this case would be controlling filters for displayed data.

Example:

Outside the HOC you can define the reducer:

```js
const DEFAULT_STATE = {
	showArchivedDates: false,
	showExpiredDates: false,
	showArchivedTickets: false,
	showExpiredTickets: false,
}

const filterReducer = ( state, action ) => {
	switch ( action.type ) {
		case 'SHOW_ARCHIVED_DATES':
			return { ...state, showArchivedDates: ! state.showArchivedDates };
		case 'SHOW_EXPIRED_DATES':
			return { ...state, showExpiredDates: ! state.showExpiredDates };
		case 'SHOW_ARCHIVED_TICKETS':
			return { ...state, showArchivedTickets: ! state.showArchivedTickets };
		case 'SHOW_EXPIRED_TICKETS':
			return { ...state, showExpiredTickets: ! state.showExpiredTickets };
	}
	throw new Error( sprintf( 'The action %s has no handling implemented', action.type ) );
};
```
Then inside this HOC you would define the following instead of the `useState` calls:

```js
const [ filterState, dispatch ] = useReducer( filterReducer, DEFAULT_STATE );
```

Then you'd just pass through `dispatch` to the wrapped component (or you could name it something like `dispatchFilterState`).  Then in components implementing this you'd just do something like `dispatch( { type: 'SHOW_ARCHIVED_DATES' } );` (or if you rename, `dispatchFilterState`) on whatever toggles the action.

Advantages:

- keep state managed in a single object
- less props passed through (just a single dispatch action)
- you could move the logic for how the `show*` value is determined via passed in arguments to the dispatch (for example you could pass arguments on the action object that help calculate the state change.
- allows for easier updating of additional filter state down the road without having to add additional props everywhere.

Reply:
me likey